### DISCUSSION TITLE
ESP32 with original Buderus KM271

### DISCUSSION THREAD

**Discussion by Andreas578** [dewenni/ESP_Buderus_KM271#121] [Q&A]
Hallo, 
ich möchte gerne "Option 2" nachbauen. Ist hierfür auch ein Schalplan verfügbar ? Sind zusätzliche Bauteile auf der Lochrasterplatine notwendig, insbesondere für die Spannungsversorgung. Ich habe bei "Option" 1 eine Spannungsanpassung von 5V auf 3,3V mittels Vorwiderstand gelesen?

**Comment by dewenni** [comment]
Hallo Andreas,
bei der Option 2 brachst du eigentlich nur den ESP mit dem MAX3232 verbinden.

```
(ESP32)GPIO17/TXD2  -> (MAX3232)TXD -> (serial cable) -> (KM271-SUBD)PIN2:RXD
(ESP32)GPIO16/RXD2 <- (MAX3232)RXD <- (serial cable) <- (KM271-SUBD)PIN3:TXD
(ESP32)GND <-> (MAX3232)GND <-> (serial cable) <-> (KM271-SUBD)PIN5:GND
```

Als Spannungsversorgung für den MAX3232 kannst du dann 3,3V oder 5V vom ESP nehmen.
Die sind in der Regel 3V / 5V Kompatibel.
Achte nur darauf, dass das serielle Kabel gekreuzt ist.

**Reply by d3ltoroxp** [reply]
Ist der vom Umfang und der Steuerung her, genau so wie mit Option 1 und dem fertigen board ?
